### PR TITLE
Send fuel tax revenue to slawer safe

### PR DESCRIPTION
**Pull request by sarahsturgeon**

### CODE REVIEW

File: lua/simfphys_darkrp_modules/fuelprice/modules/taxes/server/sv_main.lua [hunk 1]
```diff
@@ -37,6 +37,22 @@ hook.Add( "Slawer.UpdateTaxes", "CheckForUpdatedFuelTaxes", function( ply, taxDa
     FuelPrices:UpdateTaxRates( taxData.fuelTaxes )
 end )
 
+hook.Add( "SimfPhys_FuelTaxes_ChargedCustomer", "SendTaxesToSlawer", function( pump, customer, priceStruct )
+    local taxAmount = pump:GetNWFloat( "FuelTax" )
+    local price = priceStruct.price or 0
+    local taxRevenue = price / ( 1 + taxAmount )
```
Comment (samuelWilliams99) [inline lua/simfphys_darkrp_modules/fuelprice/modules/taxes/server/sv_main.lua]: Assuming the price passed into this function is post tax, tax revenue here will actually be price pre tax. I think you need `price - ( price / (taxAmount + 1 ) )` or something like that.

File: lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua [hunk 2]
```diff
@@ -91,20 +91,35 @@ function FuelPrices:AddPumpExtensions( pump )
...
+            if hookResult == false then return end
+
+            local finalPrice = priceStruct.price
 
             local formattedPrice = DarkRP.formatMoney( math.Round( finalPrice, 2 ) )
 
             local message = "You've been charged " .. formattedPrice .. " for fuel"
             DarkRP.notify( customer, 0, 5, message)
 
             customer:addMoney( -finalPrice )
```
Comment (samuelWilliams99) [inline lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua:120]: Might wanna round money before taking from client? Not sure tho

File: lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua [hunk 3]
```diff
@@ -91,20 +91,35 @@ function FuelPrices:AddPumpExtensions( pump )
...
+
+            local priceStruct = {
+                price = price
+            }
+
+            function priceStruct:SetPrice( newPrice )
+                priceStruct.price = newPrice
+            end
+
+            local hookResult = hook.Run( "SimfPhys_FuelTaxes_WillChargeCustomer", pump, customer, priceStruct )
```
Comment (samuelWilliams99) [inline lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua]: Might be better as ShouldChargeCustomer, "will" seems weird here

**Review by samuelWilliams99 — CHANGES_REQUESTED** [review]

File: lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua (hunk 2 above)
Comment (sarahsturgeon) [reply lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua:120]: NO FREE FUEL

File: lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua (hunk 3 above)
Comment (sarahsturgeon) [reply lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua]: It's an event that lets you cancel the default, or modify the price, that's why I went with "Will"

"I will charge this pump's customer with these values, does anyone have something to say about that?"

**Comment by samuelWilliams99** [comment]
G o s h ceil it then lol

File: lua/simfphys_darkrp_modules/fuelprice/modules/taxes/server/sv_main.lua (hunk 1 above)
Comment (sarahsturgeon) [reply lua/simfphys_darkrp_modules/fuelprice/modules/taxes/server/sv_main.lua]: Ah, yep

File: lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua (hunk 2 above)
Comment (sarahsturgeon) [reply lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua:120]: To elaborate on my previous statement; the customer's wallet is stored as a float, so subtracting floats should work just fine, they just won't see it entirely accurate to the dollar (their display is rounded instead)

File: lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua (hunk 3 above)
Comment (samuelWilliams99) [reply lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua]: I've not seen any other hook named like this, and hooks like "HUDShouldDraw" act almost exactly like this and use "should"

File: lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua (hunk 2 above)
Comment (samuelWilliams99) [reply lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua:120]: If other addons also do this then I guess it's fine

File: lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua (hunk 3 above)
Comment (sarahsturgeon) [reply lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua]: I don't like that I have to be consistent with a gross pattern like this 😒

Comment (samuelWilliams99) [reply lua/simfphys_darkrp_modules/fuelprice/sh_utils.lua]: ;)

**Review by samuelWilliams99 — APPROVED** [review]

**Pull request merged**